includes/class-newspack-newsletters.php

```diff
@@ -90,6 +101,170 @@ public function __construct() {
 		}
 	}
 
+	/**
+	 * Populate result in quickedit field
+	 * 
+	 * @param string $pagehook To check pagehook.
+	 */
+	public function enqueue_quick_edit_population( $pagehook ) {
+
+		// do nothing if we are not on the target pages.
+		if ( 'edit.php' != $pagehook ) {
+			return;
+		}
+
+		wp_enqueue_script( 'populatequickedit', plugins_url( '../dist/quickedit.js', __FILE__ ), array( 'jquery' ), '1.0', true );
+
+	}
+
+	/**
+	 * Adding column
+	 * 
+	 * @param array $column_array apassing columns.
+	 * 
+	 * @return mixed
+	 */
+	public function npublic_columns( $column_array ) {
+
+		$column_array['public'] = 'Published as a post';
+
+		return $column_array;
+	}
+
+	/**
+	 * Populate column value
+	 * 
+	 * @param string $column_name Column name.
+	 * @param int    $post_id Post ID.
+	 */
+	public function populate_columns( $column_name, $post_id ) {
+
+		// if you have to populate more that one columns, use switch().
+		switch ( $column_name ) :
+			case 'public':
+				echo ( get_post_meta( $post_id, 'is_public', true ) == '1' ) ? 'Yes' : 'No';
+				break;
+		endswitch;
+
+	}
+
+	/**
+	 * Display Make Newsletters public checkbox
+	 * 
+	 * @param string $column_name Coulumn name.
+	 * @param string $post_type Post Type.
+	 */
+	public function quick_edit_fields( $column_name, $post_type ) {
+
+		switch ( $column_name ) :
+			case 'public':
+				wp_nonce_field( 'quick_edit_npublic_nonce', 'npublic_nonce' );
+				echo '<fieldset class="inline-edit-col-right"><div class="inline-edit-col"><div class="inline-edit-group wp-clearfix">';
+				echo '<label class="alignleft">
+						<input type="checkbox" name="npublic">
```

Comment:
https://user-images.githubusercontent.com/7317227/122679136-95036e00-d19e-11eb-8f8f-db2ab3edf436.mov

Ideally this would be checked for newsletters which are already public (see video). Currently this is always unchecked by default.

Reply:
Ok, Let me retest it and update the functionality.

Reply:
On quick edit checkbox not selected issue: I have added quick edit script in src and defined it in webpack.config.js file. So, need to run npm run build to generate that JS in /dist/ folder.

Reply:
This is checked unchecked is coming from quickedit.js file which is generated in /dist folder.